apps/andi/lib/andi/input_schemas/ingestion.ex

```diff
@@ -58,32 +58,75 @@ defmodule Andi.InputSchemas.Ingestion do
 
   def changeset(changes), do: changeset(%__MODULE__{}, changes)
 
-  def changeset(ingestion, changes) do
+  def changeset(%Andi.InputSchemas.Ingestion{} = ingestion, changes) do
     changes_with_id = StructTools.ensure_id(ingestion, changes)
     source_format = Map.get(changes, :sourceFormat, nil)
 
     ingestion
-    |> cast(changes_with_id, @cast_fields, empty_values: [])
-    |> validate_required(@required_fields, message: "is required")
-    |> cast_assoc(:schema, with: &DataDictionary.changeset(&1, &2, source_format), invalid_message: "is required")
-    |> cast_assoc(:extractSteps, with: &ExtractStep.changeset/2)
-    |> cast_assoc(:transformations, with: &Transformation.changeset/2)
-    |> foreign_key_constraint(:targetDataset)
+    |> Changeset.cast(changes_with_id, @cast_fields, empty_values: [""])
+    |> Changeset.validate_required(@required_fields, message: "is required")
+    |> Changeset.cast_assoc(:schema, with: &DataDictionary.changeset(&1, &2, source_format), invalid_message: "is required")
+    |> Changeset.cast_assoc(:extractSteps, with: &ExtractStep.changeset/2)
+    |> Changeset.cast_assoc(:transformations, with: &Transformation.changeset/2)
+    |> Changeset.foreign_key_constraint(:targetDataset)
     |> validate_source_format()
     |> CadenceValidator.validate()
     |> validate_top_level_selector()
     |> validate_schema()
     |> validate_extract_steps()
   end
 
-  def changeset_for_draft(ingestion, changes) do
+  def changeset(%Ecto.Changeset{ data: %__MODULE__{} } = changeset, changes) do
```

Comment:
Allows creating a changeset from an existing changeset without grabbing the intermediate ingestion data (merging changesets)